hack/vagrant/bootstrap.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+set -eo pipefail
+shopt -s nullglob
+
+GOVERSION=1.9.2
```

Comment:
In order to make the shell script readable, we can use function to wrap some common functions. 

Let's make it clear and it's good for maintainers or contributors. Does it make senses?